src/routes/user.js

```diff
@@ -1,9 +1,10 @@
 const express = require('express');
 const userController = require('../controllers/user');
+const authorize = require('../middleware/guard');
 
 const router = express.Router();
 
 router.get('/profile', userController.getProfile);
 router.get('/:id', userController.getSingleUser);
-
+router.delete('/profile/delete', authorize, userController.deleteProfile);
```

Comment:
We already are using authentication middleware in the `app.js` file using `app.use`. if we wanna use it let's commented in in the app.js. but I advice against that.